components/form-builder/app/ClosingDateToggle.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { ToggleLeft, ToggleRight } from "@components/form-builder/icons";
+
+interface ClosingDateToggleProps {
+  isChecked: boolean;
+  setIsChecked: (isChecked: boolean) => void;
+  onLabel: string;
+  offLabel: string;
+  tabIndex?: number;
+}
+
+export const ClosingDateToggle = ({
+  isChecked,
+  setIsChecked,
+  onLabel,
+  offLabel,
+}: ClosingDateToggleProps) => {
+  const boldOn = isChecked ? "font-bold" : "font-normal";
+  const boldOff = !isChecked ? "font-bold" : "font-normal";
+
+  return (
+    <div
+      className="inline-block"
+      role="switch"
+      aria-checked={isChecked}
+      tabIndex={0}
+      onClick={() => {
+        setIsChecked(isChecked ? true : false);
+      }}
+      onKeyDown={(e) => {
+        if (e.key === "Enter" || e.key === " ") {
+          // toggle the switch to the opposite state
+          setIsChecked(isChecked ? true : false);
+          // Stop the browser "space" key default behavior of scrolling down
+          e.preventDefault();
+        }
+      }}
+    >
+      <div className="cursor-pointer whitespace-nowrap">
```

Comment:
Very minor and subjective. You could add something like `<span class="sr-only">Status</span>` to announce what is on/off when toggling. Though a user can figure this out by looking to the nearest heading (Status). So this would just make it really clear what is changed to on/off.

Reply:
Added "form status" --- makes it nicer